drivers/can/can_fake.c

```diff
@@ -46,7 +43,6 @@ DEFINE_FAKE_VOID_FUNC(fake_can_set_state_change_callback, const struct device *,
 
 DEFINE_FAKE_VALUE_FUNC(int, fake_can_get_max_filters, const struct device *, bool);
 
-#ifdef CONFIG_ZTEST_NEW_API
```

Comment:
The "fake" drivers stiil need some kind of `#ifdef` as they can be used outside the Ztest framework as well.